coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
+DM's deviations to `PEP 8`_ are motivated by consistency with the :doc:`cpp_style_guide`.
+Additional guidelines are included in this document to address specific requirements of the Data Management System.
 
-The naming conventions for LSST Python and C++ source have been defined to be as similar as the respective languages allow.
-In general, class names are ``CamelCase`` with leading uppercase; all other names in the source are ``camelCase`` with leading lowercase, except for module variables used as module global constants---which should be ``UPPERCASE_WITH_UNDERSCORES``.
-Names may be decorated with leading and/or trailing underscores.
+.. _PEP 8: http://www.python.org/dev/peps/pep-0008/
 
-.. _style-guide-py-2-1:
+.. _style-guide-py-flake8:
 
-User-defined names with double leading and trailing underscores MUST NOT be used
---------------------------------------------------------------------------------
+Code MAY be validated with flake8
+---------------------------------
 
-Names with double leading and trailing underscores are reserved by Python (e.g. ``__init__``, ``__name__``, ``__str__``).
+The flake8_ tool may be used to validate Python source code against the portion of PEP 8 adopted by Data Management.
+In addition, flake8_ statically checks Python for code errors.
+The separate `pep8-naming`_ plugin validates names according to the DM Python coding style.
 
-.. _style-guide-py-2-2:
+.. note::
+   
+   Flake8 only validates code against PEP 8 specifications, but does not check the full coding standard listed here.
 
-User defined names SHOULD NOT shadow python built-in functions
---------------------------------------------------------------
+.. _flake8: https://flake8.readthedocs.io
+.. _pep8-naming: http://pypi.python.org/pypi/pep8-naming
 
-Names which shadow a python built-in function may cause confusion for readers of the code.
-Creating a more specific identifier is suggested to avoid collisions.
-In the case of *filter*, ``filterName`` may be appropriate; for *filter objects*, something like ``filterObj`` might be appropriate.
+Flake8 installation
+^^^^^^^^^^^^^^^^^^^
 
-.. _style-guide-py-2-3:
+Linters are installable with :command:`pip`:
 
-Class names SHOULD use CamelCase with leading uppercase
--------------------------------------------------------
+.. code-block:: bash
 
-Python class names should follow the same conventions as C++ class names---they should be ``CamelCase`` with leading uppercase.
-Note that exceptions are classes and thus follow the same convention.
+   pip install flake8
+   pip install pep8-naming
 
-.. _style-guide-py-2-4:
+Flake8 command line invocation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Exception names relating to errors SHOULD include the suffix Error
-------------------------------------------------------------------
+.. code-block:: bash
 
-An error exception name SHOULD include the suffix ``Error``.
+   flake8 --ignore=E133,E226,E228,N802,N803 --max-line-length=110 {{python_dir}}
```

Comment:
we should mention at this point that the specific error code meanings will be discussed later on. Otherwise you immediately have a question about what E133 means. The alternative is to bring the PEP8 exceptions list forward so it appears before the `flake8` subsection. You then provide the exceptions and then say how you can check for them.